src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,25 @@
+package duke.command;
+
+import duke.Storage;
+import duke.exception.DukeException;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents an executable command.
+ */
+public abstract class Command {
+    protected boolean isExit = false;
+
+    /**
+     * Returns true if the command triggers termination of the program.
+     */
+    public boolean isExit() {
```

Comment:
It may not be necessary to store the `isExit` as an instance attribute - any child classes that will terminate the Duke application can just override `isExit` to return `true` (otherwise `false` by default).